vllm/engine/async_llm_engine.py

```diff
@@ -161,12 +169,16 @@ def get_new_and_finished_requests(self) -> Tuple[List[Dict], Set[str]]:
             self._request_streams[stream.request_id] = stream
             new_requests.append(new_request)
 
-        self.new_requests_event.clear()
-
         return new_requests, finished_requests
 
-    async def wait_for_new_requests(self):
-        await self.new_requests_event.wait()
+    async def wait_for_new_requests(self, clear: bool):
+        if not self.has_new_requests():
+            await self.new_requests_event.wait()
+        if clear:
+            self.new_requests_event.clear()
```

Comment:
Why don't we always clear this flag?
```suggestion
    async def wait_for_new_requests(self):
        if not self.has_new_requests():
            await self.new_requests_event.wait()
            self.new_requests_event.clear()
```

Reply:
Also, what's the reason behind this change? Why do we need to move the `clear` call from `get_new_and_finished_requests` to here?

Reply:
Yes, we can always clear it.

The reason is to ensure the event is cleared as soon as we have new requests